src/sbm_gwf_model.jl

```diff
@@ -294,26 +348,33 @@ function initialize_sbm_gwf_model(config::Config)
     # setup subdomains for the land and river kinematic wave domain, if nthreads = 1
     # subdomain is equal to the complete domain
     toposort = topological_sort_by_dfs(graph)
-    toposort_riv = topological_sort_by_dfs(graph_riv)
-    index_pit_land = findall(x -> x == 5, ldd)
-    index_pit_river = findall(x -> x == 5, ldd_riv)
-    streamorder = stream_order(graph, toposort)
-    min_streamorder_land = get(config.model, "min_streamorder_land", 5)
-    subbas_order, indices_subbas, topo_subbas = kinwave_set_subdomains(
-        graph,
-        toposort,
-        index_pit_land,
-        streamorder,
-        min_streamorder_land,
-    )
-    min_streamorder_river = get(config.model, "min_streamorder_river", 6)
-    subriv_order, indices_subriv, topo_subriv = kinwave_set_subdomains(
-        graph_riv,
-        toposort_riv,
-        index_pit_river,
-        streamorder[index_river],
-        min_streamorder_river,
-    )
+    if land_routing == "kinematic-wave" || river_routing == "kinematic-wave"
+        streamorder = stream_order(graph, toposort)
+    end
+    if land_routing == "kinematic-wave"
+        toposort = topological_sort_by_dfs(graph)
+        index_pit_land = findall(x -> x == 5, ldd)
+        min_streamorder_land = get(config.model, "min_streamorder_land", 5)
+        subbas_order, indices_subbas, topo_subbas = kinwave_set_subdomains(
+            graph,
+            toposort,
+            index_pit_land,
+            streamorder,
+            min_streamorder_land,
+        )
+    end
+    if river_routing == "kinematic-wave"
+        min_streamorder_river = get(config.model, "min_streamorder_river", 6)
+        toposort_riv = topological_sort_by_dfs(graph_riv)
+        index_pit_river = findall(x -> x == 5, ldd_riv)
+        subriv_order, indices_subriv, topo_subriv = kinwave_set_subdomains(
+            graph_riv,
+            toposort_riv,
+            index_pit_river,
+            streamorder[index_river],
+            min_streamorder_river,
+        )
+    end
```

Comment:
Just to check: this code is slightly different from the approach in `sbm_model.jl` (where this is in my opinion written in a clearer way). Is that correct?

Reply:
Good catch! Will align `sbm_model.jl` with this.